examples/Protobuf/messages/CMakeLists.txt

```diff
@@ -1,7 +1,41 @@
+# Example for how to compile .proto files into a library
+# Both the 'Module Compatible' and new way are shown
 
-# required since no linking
-include_directories(${PROTOBUF_INCLUDE_DIR})
-PROTOBUF_GENERATE_CPP(MESSAGES_SRC MESSAGES_HEADER messages.proto)
-add_library(messages ${MESSAGES_HEADER} ${MESSAGES_SRC})
+set(MESSAGES_PROTOS messages.proto)
+set(MESSAGES_SRCS)
+set(MESSAGES_HDRS)
 
+#Code Generation
+if(protobuf_MODULE_COMPATIBLE) #Legacy Support
```

Comment:
Let's create two examples. It will be easier to understand. Also users following new trends may not be interested in this legacy stuff at all. We already did this for OpenSSL: https://github.com/ruslo/hunter/blob/master/examples/OpenSSL/CMakeLists.txt https://github.com/ruslo/hunter/blob/master/examples/OpenSSL-imported/CMakeLists.txt